scripts/deploy.sh

```
@@ -9,12 +9,31 @@ echo "==========================================="
 NETWORK=${NETWORK:-testnet}
 ENV_FILE=".env"
 ENV_LOCAL_FILE=".env.local"
+DRY_RUN=${DRY_RUN:-false}
+
+# Mainnet safeguard
+if [ "$NETWORK" == "mainnet" ]; then
+    echo "⚠️  WARNING: You are about to deploy to STELLAR MAINNET"
+    echo "This involves real XLM and cannot be undone."
+    echo ""
+    read -p "Type 'CONFIRM' to proceed with mainnet deployment: " confirmation
```

Shellcheck (0.11.0)

[info] read without -r will mangle backslashes.

(SC2162)